lib/obsolete.gd

```diff
@@ -479,6 +479,30 @@ DeclareObsoleteSynonym( "MutableIdentityMat", "IdentityMat", "4.8" );
 ##
 DeclareObsoleteSynonym( "MutableNullMat", "NullMat", "4.8" );
 
+#############################################################################
+##
+#F  IsSemilatticeAsSemigroup( <S> ) is the semigroup <S> a semilattice
+##
+##  <#GAPDoc Label="IsSemilatticeAsSemigroup">
+##  <ManSection>
+##  <Prop Name="IsSemilatticeAsSemigroup" Arg='S'/>
+##
+##  <Description>
+##    <C>IsSemilatticeAsSemigroup</C> returns <K>true</K> if the semigroup
+##    <A>S</A> is a semilattice and <K>false</K> if it is not. <P/>
+##
+##    A semigroup is a <E>semilattice</E> if it is commutative and every
+##    element is an idempotent. The idempotents of an inverse semigroup form a
+##    semilattice.
+##
+##    This is identical to <Ref Prop="IsSemilattice" BookName = "Semigroups"/> #
+##    and is present in &GAP;&nbsp;4.8 #  only for the sake of compatibility with
+##    beta-releases.  #  It should <E>not</E> be used in new code.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareObsoleteSynonym( "IsSemilatticeAsSemigroup", "IsSemilattice", "4.8" );
```

Comment:
Also, before:
```
gap> HasIsSemilatticeAsSemigroup;
<Filter "Tester(IsSemilatticeAsSemigroup)">
```
after:
```
gap> HasIsSemilatticeAsSemigroup;
Error, Variable: 'HasIsSemilatticeAsSemigroup' must have a value
not in any function at line 4 of *stdin*
```
This is resolved by using `DeclareObsoleteSynonymAttr`, but does not resolve the problem of breaking `InstallMethod` calls.

Of course, perhaps nobody is installing methods with this filter, nor using `HasIsSemilatticeAsSemigroup` or `SetIsSemilatticeAsSemigroup`. Perhaps we in fact do not want them to use those -- so in that case, we *want* them to get an error. Also fine by me, I just want this to be a concious decision, not an accident :)